schema/site.schema.json

```
@@ -2641,6 +2641,49 @@
           "perUserDailyLimit": 100
         }
       ]
+    },
+    "autocomplete": {
+      "description": "Configuration for the autocomplete service.",
+      "type": "object",
+      "properties": {
+        "model": {
+          "description": "The model used for code completion. If using the default provider 'sourcegraph', a reasonable default model will be set.",
+          "type": "string"
+        },
+        "modelMaxTokens": {
+          "description": "The maximum number of tokens to use as client when talking to completionModel. If not set, clients need to set their own limit.",
+          "type": "integer"
+        },
+        "accessToken": {
+          "description": "The access token used to authenticate with the external completions provider. If using the default provider 'sourcegraph', and if 'licenseKey' is set, a default access token is generated.",
+          "type": "string"
+        },
+        "provider": {
+          "type": "string",
+          "description": "The external autocomplete provider. Defaults to 'sourcegraph'.",
+          "default": "anthropic",
+          "enum": ["anthropic", "openai", "sourcegraph", "azure-openai"]
+        },
+        "endpoint": {
+          "type": "string",
+          "description": "The endpoint under which to reach the provider. Currently only used for provider types \"sourcegraph\", \"openai\" and \"anthropic\". The default values are \"https://cody-gateway.sourcegraph.com\", \"https://api.openai.com/v1/chat/completions\", and \"https://api.anthropic.com/v1/complete\" for Sourcegraph, OpenAI, and Anthropic, respectively."
+        },
+        "perUserDailyLimit": {
+          "description": "If > 0, enables the maximum number of code completions requests allowed to be made by a single user account in a day. On instances that allow anonymous requests, the rate limit is enforced by IP.",
+          "type": "integer",
+          "default": 0
+        }
+      },
+      "examples": [
+        {
+          "enabled": true,
+          "model": "code-completion-model",
+          "completionModel": "code-completion",
+          "accessToken": "abc123",
+          "provider": "openai",
+          "perUserDailyLimit": 100
```

```suggestion
          "perUserDailyLimit": 0
```

avoid the potential 'I rate limited us but didn't realize it' footgun

---

thats just the example matches the completions example so I think it's ok